vcd/resource_vcd_org_user_test.go

```diff
@@ -0,0 +1,315 @@
+			debugPrintf("#[DEBUG] UPDATED CONFIGURATION: %s", configTextUpdated)
+			resourceName := "vcd_org_user." + ud.name
+			resource.Test(t, resource.TestCase{
+				PreCheck:     func() { testAccPreCheck(t) },
+				Providers:    testAccProviders,
+				CheckDestroy: testAccCheckVcdUserDestroy(ud.name),
+				Steps: []resource.TestStep{
+					resource.TestStep{
+						Config: configText,
+						Check: resource.ComposeTestCheckFunc(
+							resource.TestCheckResourceAttr(
+								resourceName, "name", ud.name),
+							resource.TestCheckResourceAttr(
+								resourceName, "role", ud.roleName),
+							resource.TestCheckResourceAttr(
+								resourceName, "provider_type", params["ProviderType"].(string)),
+							resource.TestCheckResourceAttr(
+								resourceName, "is_enabled", fmt.Sprintf("%v", params["IsEnabled"].(bool))),
+							resource.TestCheckResourceAttr(
+								resourceName, "email_address", params["EmailAddress"].(string)),
+							resource.TestCheckResourceAttr(
+								resourceName, "full_name", params["FullName"].(string)),
+							resource.TestCheckResourceAttr(
+								resourceName, "description", params["Description"].(string)),
+							resource.TestCheckResourceAttr(
+								resourceName, "instant_messaging", params["IM"].(string)),
+							resource.TestCheckResourceAttr(
+								resourceName, "deployed_vm_quota", fmt.Sprintf("%d", params["DeployedVmQuota"].(int))),
+							resource.TestCheckResourceAttr(
+								resourceName, "stored_vm_quota", fmt.Sprintf("%d", params["StoredVmQuota"].(int))),
+						),
+					},
+					resource.TestStep{
+						Config: configTextUpdated,
+						Check: resource.ComposeTestCheckFunc(
+							resource.TestCheckResourceAttr(
+								resourceName, "role", ud.secondRole),
+							resource.TestCheckResourceAttr(
+								resourceName, "full_name", updateParams["FullName"].(string)),
+							resource.TestCheckResourceAttr(
+								resourceName, "description", updateParams["Description"].(string)),
+							resource.TestCheckResourceAttr(
+								resourceName, "deployed_vm_quota", fmt.Sprintf("%d", updateParams["DeployedVmQuota"].(int))),
+							resource.TestCheckResourceAttr(
+								resourceName, "stored_vm_quota", fmt.Sprintf("%d", updateParams["StoredVmQuota"].(int))),
+						),
+					},
+					resource.TestStep{
+						ResourceName:      resourceName + "-import",
+						ImportState:       true,
+						ImportStateVerify: true,
+						ImportStateIdFunc: importStateIdByOrgUser(testConfig, ud.name),
+						// These fields can't be retrieved from user data
+						ImportStateVerifyIgnore: []string{"take_ownership", "password", "password_file"},
+					},
+				},
+			})
+		}
+	}
+	if willSkipTests {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+	cleanUserData(t)
+}
+
+func testAccCheckVcdUserDestroy(userName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*VCDClient)
+		adminOrg, err := conn.GetAdminOrg(testConfig.VCD.Org)
```

Comment:
Can we run this test with both sys admin and org admin?

Reply:
Yes. Same as we did in go-vcloud-director